s3/pkg/service/object.go

```diff
@@ -630,8 +630,13 @@ func (s *s3Service) CopyObject(ctx context.Context, in *pb.CopyObjectRequest, ou
 	targetObject.StorageMeta = res.Meta
 	targetObject.Location = targetBackendName
 	targetObject.TenantId = tenantId
+	targetObject.UserId = srcBucket.UserId
```

Comment:
Here, userid can be get from ctx, see line 521, not from source bucket.

Reply:
@sfzeng Done , Thanks